src/miner.cpp

```diff
@@ -171,12 +171,14 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     const staking::ActiveChain *active_chain = GetComponent<staking::ActiveChain>();
     for (const staking::Coin &coin : stakeable_coins) {
       proposer::EligibleCoin eligible_coin = {
-          coin,
-          GetRandHash(),
+          staking::Coin(active_chain->GetBlockIndex(coin.GetTransactionId()),
+              COutPoint(coin.GetTransactionId(), coin.GetOutputIndex()),
+              CTxOut(coin.GetAmount(), scriptPubKeyIn)),
+          GetRandHash(), //TODO UNIT-E: At the moment is not used, since we still have PoW here
           GetComponent<blockchain::Behavior>()->CalculateBlockReward(nHeight),
-          0,
-          0,
-          0
+          0, //TODO UNIT-E: At the moment is not used, since we still have PoW here
+          0, //TODO UNIT-E: At the moment is not used, since we still have PoW here
+          0 //TODO UNIT-E: At the moment is not used, since we still have PoW here
```

Comment:
Maybe it's not used, but we do not have PoW here.

Reply:
I just reverted some changes from previous commit here. Basically this stuff goes away in the next PR where I get rid of the `CreateNewBlock` from the tests too.